sdk/identity/azure-identity/azure/identity/_internal/shared_token_cache.py

```diff
@@ -88,13 +89,22 @@ class SharedTokenCacheBase(ABC):
     def __init__(self, username=None, **kwargs):  # pylint:disable=unused-argument
         # type: (Optional[str], **Any) -> None
 
-        authority = kwargs.pop("authority", None)
-        self._authority = normalize_authority(authority) if authority else get_default_authority()
-
-        environment = urlparse(self._authority).netloc
-        self._environment_aliases = KNOWN_ALIASES.get(environment) or frozenset((environment,))
-        self._username = username
-        self._tenant_id = kwargs.pop("tenant_id", None)
+        self._auth_record = kwargs.pop("authentication_record", None)  # type: Optional[AuthenticationRecord]
+        if self._auth_record:
+            # authenticate in the tenant that produced the record unless 'tenant_id' specifies another
+            authenticating_tenant = kwargs.pop("tenant_id", None) or self._auth_record.tenant_id
+            self._tenant_id = self._auth_record.tenant_id
+            self._authority = self._auth_record.authority
+            self._username = self._auth_record.username
+            self._environment_aliases = frozenset((self._authority,))
+        else:
+            authenticating_tenant = "common"
```

Comment:
If we're using the Azure CLI client id I believe we should be defaulting to the 'organizations` endpoint rather than `common`.

Reply:
I kept `common` to maintain the existing behavior. That was probably too conservative given I don't know a scenario the change would break. So, I've switched to `organizations`.